### PR TITLE
IFTAQCAB-31 (Add new group) + IFTAQCAB-33 (Delete group)

### PR DESCRIPTION
**Pull request by promahos**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/pageobjects/modals/AddingFormModal.java [hunk 1]
```diff
@@ -3,22 +3,45 @@
...
     @Step("Adding form modal: Set {value} for {field}")
     public AddingFormModal setValueFor(AddingFormFields field, String value) {
-        field
+        var fieldElement = field
                 .getName()
-                .should(appear, ofSeconds(5))
-                .sendKeys(value);
+                .should(appear, ofSeconds(5));
+
+        if (Objects.requireNonNull(fieldElement
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/modals/AddingFormModal.java:21]: static import of "requireNonNull"

extract into variable "fieldElement.getAttribute("class")"

File: src/main/java/com/softserveinc/ita/pageobjects/modals/AddingFormModal.java [hunk 2]
```diff
@@ -3,22 +3,45 @@
...
+
+            fieldElement
+                    .shouldBe(focused)
+                    .sendKeys(value);
+        }
 
         return this;
     }
+
+
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/modals/AddingFormModal.java:44]: delete extra new line

File: src/main/java/com/softserveinc/ita/steps/GroupsStep.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public class GroupsStep {
+    private GroupsPage page;
+
+    public void openPage() {
+        page = new LoginPage()
+                .login(ADMIN_LOGIN, ADMIN_PASSWORD)
+                .openGroupsPage();
+    }
+
+    public void addNewGroup(GroupEntity group) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/steps/GroupsStep.java:23]: extract into variable 

```
group
                        .getSpeciality()
                        .getName()
```

Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/steps/GroupsStep.java:23]: also this

```
group
                        .getFaculty()
                        .getName()
```

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/models/GroupEntity.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+@Builder
+@Getter
+@Setter
+public class GroupEntity {
+    private String name;
+    private SpecialityEntity speciality;
+    private FacultyEntity faculty;
+
+    public static String getNewValidName() {
```
[the remark this answers is not in the document]
Comment (promahos) [reply src/main/java/com/softserveinc/ita/models/GroupEntity.java:17]: changed similar to SpecialityEntity in order to get already a whole object, not just a string
I'm not sure about moving this to RandomUtil, as it seems to me, that logic, what name is considered to be valid (or not) should be inside the entity class

**Review by Nikodych — APPROVED** [review]

**Pull request merged**